### ISSUE TITLE
Cannot start jackdbus on kernel 4.4 in duplex or capture-only mode (internal soundcard on a mid-2010 MacBook Pro)

### ISSUE THREAD

**Issue by jmaibaum** [jackaudio/jack2#186]
On Arch Linux x86-64, running with Linux kernel 4.4.1, pulseaudio 8.0, and jack 1.9.10, I cannot start JACK anymore in duplex or capture-only mode using the internal soundcard from a mid-2010 MacBook Pro. Playback-only mode works.

When downgrading to Linux kernel 4.3.3, JACK starts normally in all configurations, as expected.

Trying to start jack with the `jack_control` command (using alsa as driver), I get the following messages after the timeout:

    $ jack_control start
    --- start
    ERROR:dbus.proxies:Introspect error on :1.52:/org/jackaudio/Controller: dbus.exceptions.DBusException: org.freedesktop.DBus.Error.NoReply: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
    DBus exception: org.freedesktop.DBus.Error.NoReply: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.

Qjackctl is a little more verbose in its output. With the same configuration, I get the following log messages:

    Sun Feb  7 15:02:32 2016: Starting jack server...
    Sun Feb  7 15:02:32 2016: JACK server starting in realtime mode with priority 10
    Sun Feb  7 15:02:32 2016: self-connect-mode is "Don't restrict self connect requests"
    Sun Feb  7 15:02:32 2016: Acquired audio card Audio0
    Sun Feb  7 15:02:32 2016: creating alsa driver ... hw:NVidia|hw:NVidia|512|2|48000|0|0|nomon|swmeter|-|32bit
    Sun Feb  7 15:02:33 2016: configuring for 48000Hz, period = 512 frames (10.7 ms), buffer = 2 periods
    Sun Feb  7 15:02:33 2016: ALSA: final selected sample format for capture: 32bit integer little-endian
    Sun Feb  7 15:02:33 2016: ALSA: use 2 periods for capture
    Sun Feb  7 15:02:33 2016: ALSA: final selected sample format for playback: 32bit integer little-endian
    Sun Feb  7 15:02:33 2016: ALSA: use 2 periods for playback
    Sun Feb  7 15:02:38 2016: ERROR: JackPosixProcessSync::LockedTimedWait error usec = 5000000 err = Connection timed out
    Sun Feb  7 15:02:38 2016: ERROR: Driver is not running
    Sun Feb  7 15:02:38 2016: ERROR: Cannot open client name = dbusapi
    Sun Feb  7 15:02:38 2016: ERROR: failed to create dbusapi jack client
    Sun Feb  7 15:02:38 2016: ERROR: Unknown request 4294967295
    Sun Feb  7 15:02:38 2016: ERROR: CheckSize error size = 0 Size() = 12
    Sun Feb  7 15:02:38 2016: ERROR: CheckRead error

Using the classic `jackd` command instead of the DBus interface, the same error message as above appears as soon as one tries to connect any client to the JACK server.

Using an USB audio interface, I don't face any issues. All modes work as expected.

I also reported this issue in Arch Linux's bug tracker, see https://bugs.archlinux.org/task/48073 , where someone else was able to reproduce. I was given the advice to also report here, just to make sure.

**Comment by mseeber** [comment]
Sun Feb  7 15:02:32 2016: creating alsa driver ... hw:NVidia|hw:NVidia|512|2|48000|0|0|nomon|swmeter|-|32bit

I am most certainly you do not intend to use the NVIDIA card as your audio interface?

Select your soundcard before starting jack. This wasn't necessary before, because your older kernel did assign the right soundcard to index 0, now the new kernel assigns the NVIDIA card to index 0.

**Comment by jmaibaum** [comment]
On this mid-2010 MacBook Pro, the internal soundcard is actually called NVidia:

    $ cat /proc/asound/cards 
     0 [NVidia         ]: HDA-Intel - HDA NVidia
                          HDA NVidia at 0xd3480000 irq 20

It's been called like this since I've installed Arch Linux.

So, I'm pretty sure that the correct card is selected when trying to start jack. Or do you mean something else?

**Comment by mseeber** [comment]
OK, a MacBook...
Does the soundcard work with pulseaudio and ALSA?

**Comment by jmaibaum** [comment]
Yes, the soundcard works with both PulseAudio and ALSA. I might emphasize that, as mentioned above, JACK used to work perfectly, too, until I did the kernel update to 4.4.

**Comment by mseeber** [comment]
I have read inyour previous threads/reports that jack starts properly with the dummy driver, is that still correct?

I have listed some invocations for jackd which you can try, so wen can see where it fails:

    jackd -d dummy

    #these will most likely fail
    jackd -d  alsa
    jackd -d alsa -d hw:0

    #playback only
    jackd -d alsa -P hw:0
    #capture only
    jackd -d alsa -C hw:0

**Comment by jmaibaum** [comment]
As I have installed the dbus version of JACK 2, I don't have a jackd command. I tried to reformulate your commands to the with jack_control commands and can report the following:

    $ jack_control ds dummy
    --- driver select "dummy"
    $ jack_control start
    --- start
    $ jack_control status
    --- status
    started
    $ jack_control stop
    --- stop
    $ jack_control ds alsa
    --- driver select "alsa"
    $ jack_control start
    --- start
    DBus exception: org.freedesktop.DBus.Error.NoReply: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
    $ jack_control status
    --- status
    ERROR:dbus.proxies:Introspect error on :1.215:/org/jackaudio/Controller: dbus.exceptions.DBusException: org.freedesktop.DBus.Error.NoReply: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
    DBus exception: org.freedesktop.DBus.Error.NoReply: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
    $ jack_control stop
    --- stop
    ERROR:dbus.proxies:Introspect error on :1.215:/org/jackaudio/Controller: dbus.exceptions.DBusException: org.freedesktop.DBus.Error.NoReply: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
    DBus exception: org.freedesktop.DBus.Error.NoReply: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.

All of the DBus (error)messages above only appeared after a long timeout delay. I had to `killall -9 jackdbus` to make the jack_control command work again.

I then issued a `jack_control dps device hw:0` before trying to start jack again, but that gave the same results.

Unfortunately, I don't know how to tell jack_control to start in playback-only or capture-only mode, so I have not tested that, yet.

**Comment by jmaibaum** [comment]
Ok, I just installed the "jack2" package from ArchLinux which provides the classic jackd command as well. Now trying your suggested invocations, *all those commands fail, except the playback only version,* which works! I did then start PureData in JACK mode to have something to feed sound into JACK (as *PulseAudio did not show its JACK sink*, what it used to do when JACK started) and I was able to hear sound as expected.
When using any other invocation that includes the capture mode, it fails with the same Error messages as in my first post, *as soon as one tries to connect any client to the server.*

When I connect my USB audio interface, it seems to have no problems whatsoever. It works in all configurations. So, I will reword my issue report above.

**Comment by mseeber** [comment]
So duplex and capture do not work, sounds similar to #116 which is still present on my machine, a Lenovo laptop also with an Intel HDA but not with Nvidia components.

**Comment by jmaibaum** [comment]
Reading through #116 I think this might be related. What makes me wonder though is that I had no troubles with my internal soundcard until the kernel 4.4 update, i.e. duplex mode was working with HDA Intel/NVidia the whole time from at least kernel 4.1 (that's when I seriously started using Arch Linux on this machine, and started to work with jack regularly). In fact, I still have a 4.1-realtime kernel installed that I can use.

grep'ing through the 4.4 changelog, I see some driver changes for the HDA Intel driver. Maybe I should file a bug report with the ALSA devs then? Or did you do that already for your issue?

**Comment by mseeber** [comment]
Yes, please report the bug to the ALSA devs.
You can reference these reports and your other reports, maybe that helps. If there is anything to test or add, I may follow up on the mailing lists.

**Comment by jmaibaum** [comment]
To keep you informed, I just tried to file a bug at the ALSA project, but their bug tracker (https://bugtrack.alsa-project.org/alsa-bug) is down at the moment. Will see when I next find time to check again...

**Comment by mseeber** [comment]
I think they will be more responsive on the mailing list. Kernel folks usually do not like bug trackers that much, which would explain why this thing is down.

**Comment by jmaibaum** [comment]
I just posted a message to the alsa-users list: http://article.gmane.org/gmane.linux.alsa.user/39426

**Comment by jmaibaum** [comment]
Are there any other things I could do to debug the issue from the JACK side? Even if it's a driver issue (the ALSA guys did not respond yet), it occurs only when I try to start JACK. Every other non-JACK audio application works without a problem.

**Comment by jmaibaum** [comment]
I can report some news. It seems by now that the bug was *not* due to some regression in the kernel and/or jack, but due to some misconfigurations on my side. Last weekend, I have tried various different live Linux distributions with a recent kernel (4.3 and 4.4 were among them) and all of them worked.

So, at one point I decided to give it a go and do a full reinstall of my Arch System and sure enough, the problem was gone. I have since then tested both jack1 and jack2 (with both the old jackd and the jackdbus interface) and all of them seem to work as expected in all modes.

I am really curious what was wrong configuration-wise before the reinstall. But, at the moment it seems that there never was a real bug, only stupid misconfiguration introduced by me. I will keep observing jack and see if the problem reoccurs at one point, as I am slowly reinstalling packages that I had been using before the reinstall, but until now I have no problems any more.
